mne/decoding/time_gen.py

```diff
@@ -817,6 +817,172 @@ def _time_gen_one_fold(clf, X, y, train, test, scoring):
+
+        Default: 'cross-validation'
+    n_jobs : int
+        Number of jobs to run in parallel. Defaults to 1.
+
+    Attributes
+    ----------
+    y_train_ : np.ndarray, shape (n_samples,)
+        The categories used for training.
+    estimators_ : list of list of sklearn.base.BaseEstimator subclasses.
+        The estimators for each time point and each fold.
+    y_pred_ : np.ndarray, shape (n_train_times, n_test_times, n_epochs,
+                                 n_prediction_dims)
+        Class labels for samples in X.
+    scores_ : list of lists of float
+        The scores (mean accuracy of self.predict(X) wrt. y.).
+        It's not an array as the testing times per training time
+        need not be regular.
+    test_times_ : dict
+        The same structure as ``train_times``.
+    cv_ : CrossValidation object
+        The actual CrossValidation input depending on y.
+
+    Notes
+    -----
+    The function implements the method used in:
+
+        Jean-Remi King, Alexandre Gramfort, Aaron Schurger, Lionel Naccache
+        and Stanislas Dehaene, "Two distinct dynamic modes subtend the
+        detection of unexpected sounds", PLOS ONE, 2013
+
+    .. versionadded:: 0.9.0
+
+    Returns
+    -------
+    """
+
+    def predict(self, X, test_times='diagonal', **kwargs):
+        """ Test each classifier at each time point.
+
+        Note. This function sets and updates the ``y_pred_`` and the
+        ``test_times`` attribute.
+
+        Parameters
+        ----------
+        epochs : instance of Epochs
+            The epochs. Can be similar to fitted epochs or not. See independent
+            parameter.
+        test_times : str | dict | None
+            A dict to configure the testing times.
+            If test_times = 'diagonal', test_times = train_times: decode at
+            each time point but does not generalize.
+
+            'slices' : np.ndarray, shape (n_clfs,)
+                Array of time slices (in indices) used for each classifier.
+                If not given, computed from 'start', 'stop', 'length', 'step'.
+            'start' : float
+                Time at which to start decoding (in seconds). By default,
+                min(epochs.times).
+            'stop' : float
+                Maximal time at which to stop decoding (in seconds). By
+                default, max(times).
+            'step' : float
+                Duration separating the start of to subsequent classifiers (in
+                seconds). By default, equals one time sample.
+            'length' : float
+                Duration of each classifier (in seconds). By default, equals
+                one time sample.
+
+            If None, empty dict. Defaults to None.
+        picks : np.ndarray (n_selected_chans,) | None
+            Channels to be included.
+
+        Returns
+        -------
+        y_pred_ : np.ndarray, shape (n_train_time, n_test_time, n_epochs,
+                               n_prediction_dim)
```

Comment:
this is the reason I ask. The `shape` looks weird to me. Shouldn't it be `(n_train_time, n_epochs, n_folds)` ? What is `n_prediction_dim` ?